SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -619,15 +647,23 @@ - (void)sendConnectionRequest:(__kindof SDLRPCRequest *)request withResponseHand
         return;
     }
 
-    [self sdl_sendRequest:request withResponseHandler:handler];
+    dispatch_async(_lifecycleQueue, ^{
```

Comment:
This should only change the encryption BOOL (though it will be on the request itself) to YES from NO, not visa-versa. That case should force a failure through the response handler (you will return `nil` response, the request, and an appropriate new error value).

Reply:
If the RPC requires encryption and the developer setPayloadProtected to NO then the handler would return error as ENCRYPTION_REQUIRED.   this would cause the developer to want to check whether an RPC needs encryption for each request.  that's why I was thinking this method should always check to if an RPC needs encryption from the permission manager and then change to BOOL from NO to YES

Reply:
Ah crap, I miswrote my comment. I meant that it should only change the encryption BOOL from NO to YES.

If the RPC requires encryption from the head unit side, but the developer says NO, then we can assume that the developer doesn't care if it's encrypted or not. But if they say YES, then we have to assume they want to require encryption. If the RPC requires encryption from the head unit side, then we can change it to YES for them to enable the RPC to work. If they set it to YES and we can't encrypt, then we should fail early.

I apologize for the mis-type in my first comment.